### DISCUSSION TITLE
Resolve an issue with Icaros not thumbnailing a specific filetype

### DISCUSSION THREAD

**Discussion by sanitysama** [Xanashi/Icaros#68] [Show and tell]
I've had an issue where Icaros wasn't thumbnailing .webm files for a very long time. It was difficult to troubleshoot, I checked a wide variety of system activities to no avail until discovering the solution today, so I came to share what I did to fix the problem to try to save others the struggle of figuring out why a specific file type is not thumbnailing. **_This is a rare and strange issue._** I am only sharing this here because Icaros is the only program I have or know of that can thumbnail the WEBM filetype.

**This issue was not the fault of a programming error in Icaros, and Icaros did not cause the issue.** My best guess is my registry got a little corrupted when my PC shut off unexpectedly many months ago. I noticed file associations with video files, mp4 mkv were screwed up after that point. I was being prompted which program to open them with over and over. I reset those file associations with MPC-HC and all has been working fine ever since, but I did not get WEBM thumbnailing again until today.

No changes of settings in Icaros fixed the problem. Reinstalling Icaros also had no effect. Rebuilding cache didn't help either. The problem came from somewhere else.

I checked the registry for problems and it turns out `HKEY_CLASSES_ROOT\.webm\ShellEx` was inaccessible, perhaps due to corruption.

I receive the following error attempting to access this registry key `[Window Title] Error Opening Key  [Content] ShellEx cannot be opened. An error is preventing this key from being opened. Details: Access is denied.   [OK]` This is not normal.

- To resolve this, download **[PsExec Tool](https://learn.microsoft.com/en-us/sysinternals/downloads/psexec)** from Microsoft.
- Extract its contents to `C:\Windows\System32`
- Open Command Prompt in Administrator mode: Hold WIN+R, type `cmd` and press CTRL+SHIFT+ENTER
- Verify the window title of the Command Prompt window begins with "Administrator"
- Type or paste the command **`psexec -i -d -s c:\windows\regedit.exe`**
- psexec will open Registry Editor, but it will be running under the SYSTEM account; verify by opening Task Manager, go to the Details tab, click within the list of processes, type 'regedit' and observe the "User Name" tab will say SYSTEM for regedit.exe
- Right-click the key you cannot access, for me it was ShellEx, and click **Permissions...**
- Click **Add...**
- The defaults should say Select this object type: Users, Groups, or Built-in security principals
- From this location: (This is the name of your PC)
- Within the textarea for "Enter the object names to select:" you enter your **user account name.** That would be C:\Users\\[user account name] enter this exact value.
- Click **Check Names**. It will change the formatting of your account name, this is how you know it worked and your account name is being recognized as valid.
- Click **OK** to close the window
- Under "Permissions for [user account name]" check **Full control** and **Read**.
- Click **OK**
- **Open** IcarosConfig.exe
- Go to **Tools**
- Under WINDOWS EXPLORER click **Refresh**
- Open a folder with the affected filetype, Icaros should be able to thumbnail it now.

If you close and reopen Registry Editor, you'll see some new keys got created in the folder you just fixed the permissions for.

**tl;dr:** The issue was keys could not be created in the Windows Registry under the ShellEx folder for the specific file type as the permissions were corrupt, preventing normal functionality of Icaros.

**Comment by Xanashi** [comment]
Hi sanitysama,

Thank you so much for sharing your solution! 
You're the best for taking the time to help others with the same issue, even though you didn't have to.

Problematic registry permissions is definitely one of the most common reasons why Icaros doesn't work on
some users systems. So having this guide is incredible.

One question, after fixing the permissions in Regedit, are you not required to re-enable Icaros to create the shellex keys?

**Reply by sanitysama** [reply]
All I did was open a folder full of webm files, then used IcarosConfig's tool button to refresh file explorer and it started thumbnailing them. I do notice that enabling/disabling thumbnailing also causes a file explorer refresh, so that should also work.

**Reply by Xanashi** [reply]
I'm just surprised that the shellex keys was added after a simple refresh, almost like Windows had queued up the registry changes and was waiting for the permissions to be in order. 
But if it works, I won't complain. 😄

**Comment by brokedarius** [comment]
This help me greatly in fixing my issue but if your case differ slightly like me I suggest running around the same steps: opening regedit as SYSTEM, setting up .webm permissions exactly the same as .mkv if unsure and then applying the icaros refresh as described.
